src/orquestra/sdk/_ray/_dag.py

```diff
@@ -187,23 +187,14 @@ class RayParams:
 class RayRuntime(RuntimeInterface):
     def __init__(
         self,
-        client: RayClient,
         config: RuntimeConfiguration,
         project_dir: Path,
+        client=None,
     ):
-        self._client = client
-        self._config = config
-        self._project_dir = project_dir
+        if client is None:
+            client = RayClient()
 
-        self._log_reader: LogReader = _ray_logs.DirectRayReader(
-            _services.ray_temp_path()
-        )
-
-    @classmethod
-    def from_runtime_configuration(
-        cls, project_dir: Path, config: RuntimeConfiguration, verbose: bool
-    ) -> "RuntimeInterface":
-        client = RayClient()
+        self._client = client
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏This mutates the input `client`. I think it's OK here, but this might also work:
```suggestion
        self._client = client or RayClient()
```